### DISCUSSION TITLE
DateTime ISO cached in REDIS

### DISCUSSION THREAD

**Discussion by SebaBoler** [MichalLytek/type-graphql#1304] [Q&A]
Hello,

i have issue with fields type Date with cache in Redis:
```
@Field()=> GraphQLISODateTime)
createdAt: Date
```
Additionally i have setup my graphql
```
      buildSchemaOptions: {
        dateScalarMode: 'isoDate',
      },
```

First request i see Date in ISO. Works perfect. Strange when i remove GraphQLISODateTime i will see timestamp.
Next request this same query returned only null for createdAt.

I read about serialize but cant use ISOToString() beacuse serialize return number.

Could anyone help me ?

**Comment by MichalLytek** [comment]
Redis do not store instance of `Date` but just a string/number. So you need to serialize the value on publish and deserialize when receiving in the subscription handler.

**Reply by SebaBoler** [reply]
Yeah i know Redis return string - but when type of createdAt is Date process of serialize is going to this code
```
  serialize(value: Date) {
    return value instanceof Date ? value.getTime() : null;
  }
``` 

and response is always null beacuse type of value from redis is string and serialize in Promise has only number. How return ISODate liked that ?
<img width="448" alt="Zrzut ekranu 2022-07-7 o 20 27 20" src="https://user-images.githubusercontent.com/9715474/177843632-e074fbd7-f96b-4397-be2f-0549288c44b4.png">

**Reply by MichalLytek** [reply]
You need to deserialize the payload from redis (string) into Date instance. Otherwise all your other logic will crash in runtime, like `date.getMonth is not a function`

**Reply by SebaBoler** [reply]
Have you have any examples? 
I dont know how to find method or function where i cant catch response from Redis before i will see in Apollo Studio.
Im using TypeOrm to connect with Postgresql

**Reply by MichalLytek** [reply]
https://github.com/MichalLytek/type-graphql/blob/master/examples/redis-subscriptions/recipe.resolver.ts#L68

**Reply by SebaBoler** [reply]
Yeah its works. But cant use when im using complicated response Object with related tables. 
For any response from Query/Sub i used Entity original with Date.

**Reply by MichalLytek** [reply]
This is not a TypeGraphQL question then but Redis serialization/general TS serialization issue 😉

**Reply by SebaBoler** [reply]
I resolved the problem using transformer. 
Problem was more complicated beacuse Postgresql has Date = timestamptz and Redis saved Date as IsoDate.

Thank you for your help
Best Regards
